.github/workflows/cmake.yml

```diff
@@ -0,0 +1,137 @@
+name: CMake
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    name: CMake with default option on the default GitHub Ubuntu
+    # The CMake configure and build commands are platform agnostic and should work equally
+    # well on Windows or Mac.  You can convert this to a matrix build if you need
+    # cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      - name: Install required packages
+        run: |
+          sudo apt update
+          sudo apt install libboost-all-dev librange-v3-dev
+
+      - name: Configure CMake
+        # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+        # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+        run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}}
+
+      - name: Build
+        # Build your program with the given configuration
+        run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}}
+
+      - name: Test
+        working-directory: ${{github.workspace}}/build
+        # Execute tests defined by the CMake configuration.
+        # See https://cmake.org/cmake/help/latest/manual/ctest.1.html for more detail
+        run: ctest -C ${{env.BUILD_TYPE}}
+
+  docker-gcc7-openmp:
+    name: OpenMP & OpenCL POCL with G++ 7
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      - uses: docker://ubuntu:hirsute
+        env:
+          C_COMPILER: gcc-7
+          CXX_COMPILER: g++-7
+          OPENMP: ON
+          OPENCL: ON
+        with:
+          entrypoint: /github/workspace/.github/workflows/run_ci_in_docker.bash
```

Comment:
`${{github.workspace}}` is better than `/github/workspace`, because it is more robust in case something changes on GitHub Actions side

Reply:
The problem here is that the variable environments have not the same values inside the Action and inside the Docker inside the Action.
Here I need to use what will be `${{github.workspace}}` inside Docker *before* entering Docker and I have not found how to get this.
This is why I have put in the script run inside docker some `df` and `env` to see the matching.

Reply:
Ok, I got it. From the official [docs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#docker-container-filesystem) it seems to be a documented location, i.e. it is more or less safe to hardcode it, but I'm a bit sad that there is no pre-defined env variable or something which would allow you to use that path without hardcoding it

Reply:
Yes, this looks like some oversight to me. They should provide an environment variable for these too...